src/linear_algebra/matrix.lean

```diff
@@ -813,6 +820,32 @@ begin
   apply h₂,
 end
 
+variables {V : Type*} [add_comm_group V] [vector_space K V]
+variables {v₁ : n → V} (hv₁ : is_basis K v₁)
+
+lemma ker_to_lin_eq_bot [decidable_eq n] (M : matrix n n K) (hM : M.det ≠ 0) :
+  (to_lin hv₁ hv₁ M).ker = ⊥ :=
+begin
+  rw submodule.eq_bot_iff,
+  intros x hx,
+  calc x = to_lin hv₁ hv₁ (M⁻¹ ⬝ M) x : by simp [nonsing_inv_mul M (is_unit.mk0 _ hM)]
+     ... = to_lin hv₁ hv₁ M⁻¹ (to_lin hv₁ hv₁ M x) : by simp [to_lin_mul hv₁ hv₁ hv₁]
+     ... = (to_lin hv₁ hv₁ M⁻¹) 0 : by rw mem_ker.mp hx
+     ... = 0 : linear_map.map_zero _
+end
```

Comment:
I'm confused, isn't this just `to_lin.injective` followed by `ker_eq_bot_of_injective`?

Reply:
Ah, nevermind - I'm talking about `(to_lin hv₁ hv₁).ker = ⊥`

Reply:
Looking at it again I'm inclined to agree that this should have an easier proof.